Views

Lucas-Erkana opened the pull request:
![Hi Reviewer](https://github.com/Lucash2022/Blog_App/assets/41428579/d36d75dc-3532-4527-8ecd-d7b5ed39451f)

**In this [project](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md), these tasks were completed:**

- [x] Updated UsersController's "show" action to fetch and assign the user and their recent posts. 
- [x] Updated PostsController's "index" action to correctly fetch and assign posts based on the user.
- [x]  Removed unnecessary code from PostsController's "show" action. 
- [x] Added comments for better code readability and understanding.
- [x] Updated the HTML structure to use a <main> element with appropriate id and class attributes.
- [x] Refactor users/index.html.erb view
- [x] Enhance users/show.html.erb view
- [x] Revamp posts/index.html.erb view
- [x] Update posts/show.html.erb view
- [x] Implement the design from the [wireframes](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md#project-requirements).
- [x] Apply styling

Lucas-Erkana committed:
Refactor UsersController and PostsController

•	Updated UsersController's "show" action to fetch and assign the user and their recent posts.
•	Updated PostsController's "index" action to correctly fetch and assign posts based on the user.
•	Removed unnecessary code from PostsController's "show" action.
•	Added comments for better code readability and understanding.

Lucas-Erkana committed:
Refactor users/index.html.erb view

•	Updated the HTML structure to use a <main> element with appropriate id and class attributes.
•	Replaced the simple user listing with a dynamic user list using an unordered list (<ul>) and list items (<li>).
•	Added user profile picture and user information for each user using appropriate HTML tags and Rails helpers.
•	Included a link to each user's profile page and displayed the number of posts for each user.
•	Added explanatory comments to improve code readability and understanding.

Lucas-Erkana committed:
Enhance users/show.html.erb view

•	Updated the <main> element with id "user-index" and class "page" to provide a clear structure for the content.
•	Added a "Home" button, a "Users" button, and a "Back" button to improve navigation within the application.
•	Refactored the user information section to utilize appropriate HTML tags and classes, improving the overall layout.
•	Added a "Bio" section to display the user's bio information.
•	Incorporated links to view individual posts, enabling users to access more details about each post.
•	Included a "See all posts" button to allow users to navigate to a page displaying all posts by the user.
•	Made code enhancements and added explanatory comments for better code readability.

Lucas-Erkana committed:
Revamp posts/index.html.erb view

•	Updated the <main> element with id "posts" and class "page" for proper styling and structure.
•	Added a "Home" button, a "Users" button, and a "Back" button to enhance navigation within the application.
•	Improved the user profile section by using appropriate HTML tags and classes for better presentation.
•	Refactored the code to display a list of posts for the given user, including post titles, details, comment counts, and like counts.
•	Incorporated a container to display recent comments on each post, or a message if no comments are available.
•	Made code enhancements and added explanatory comments for better code readability.

Lucas-Erkana committed:
Update posts/show.html.erb view

•	Modified the <main> element with id "post-index" and class "page" for proper styling and structure.
•	Added a "Home" button, a "Users" button, and a "Back" button to improve navigation within the application.
•	Enhanced the display of a specific post for a user, including the post title, author name, comment count, and like count.
•	Organized the post information using appropriate HTML tags and classes.
•	Incorporated a container to display comments on the post, or a message if no comments are available.
•	Made code enhancements and added explanatory comments for better code readability.

Lucas-Erkana committed:
Update application.css styles

•	Applied consistent background color and overall styling using the universal selector.
•	Created a flexbox layout for improved page structure.
•	Styled headings, links, buttons, and user info for better visual presentation.
•	Adjusted container sizes, margins, and alignments for user profiles, posts, and comments.
•	Added specific styles for user bio, post titles and details, and comment sections.
•	Implemented responsive design for larger screens using a media query.

Lucas-Erkana committed:
Remove list of users on the top of the users page

AnselemOdims commented on app/controllers/users_controller.rb:
```diff
@@ -1,9 +1,14 @@
-    @user = User.find(params[:id])
+class PostsController < ApplicationController
+  def index
+    @user = User.find(params[:user_id])
+    user_id = params[:user_id]
+    @posts = Post.where(user_id:)
+    @comments = Comment.all
   end
 
-  def index
-    @users = User.all
+  def show
+    @post = Post.find(params[:id])
+    @user = User.find(params[:user_id])
+    @comments = Comment.where(post_id: params[:id])
   end
 end
```
- [x] While you have made a great effort in this milestone, you have put the `PostsController` class in the `users_controller.rb` file which might cause some issues as Rails rely on these naming conventions to associate the controller to the correct view file.

Lucas-Erkana replied:
@AnselemOdims 
Thank you so much for picking up that error. I have rectified the mistake at this [commit.](https://github.com/Lucash2022/Blog_App/pull/9/commits/d23213d1f2cc919bcdc73fb5aeec3d73fe3d460e)

AnselemOdims commented on app/controllers/posts_controller.rb:
```diff
@@ -1,10 +1,10 @@
-    @post = Post.where(user_id: user)
+class UsersController < ApplicationController
+  def show
+    @user = User.find(params[:id])
+    @posts = @user.recent_post
   end
 
-  def show
-    @post = Post.find_by(user_id: params[:user_id])
+  def index
+    @users = User.all
   end
 end
```
- [x] While you have made a great effort in this milestone, you have put the `PostsController` class in the `users_controller.rb` file which might cause some issues as Rails rely on these naming conventions to associate the controller to the correct view file.

Lucas-Erkana replied:
@AnselemOdims 
Thank you so much for picking up that error. I have rectified the mistake at this [commit.](https://github.com/Lucash2022/Blog_App/pull/9/commits/d23213d1f2cc919bcdc73fb5aeec3d73fe3d460e)

AnselemOdims reviewed (changes requested):
### Hi @Lucash2022 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

Lucas-Erkana committed:
Refactor controller classes for correct file placement

Summary:
•	Moved the PostsController class to the posts_controller.rb file and the UsersController class to the users_controller.rb file.
•	This ensures adherence to Rails naming conventions and proper association between controllers and view files.
Details:
•	Moved PostsController class to posts_controller.rb file
•	Moved UsersController class to users_controller.rb file

Lucas-Erkana committed:
Fix linter errors

Owusu-Desmond reviewed (approved):
Hi @Lucash2022 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
📌 Linters are passing ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 README file is okay ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
